integration-tests/manifests/annotations/validate_annotations_test.go

```diff
@@ -0,0 +1,1003 @@
+	_, err := clientset.CoreV1().Namespaces().Get(context.Background(), name, metav1.GetOptions{})
+	if err == nil {
+		// Namespace exists - you can choose to return nil or delete and recreate it
+		return nil // Use this line if you want to just use the existing namespace
+		// Uncomment below lines if you want to delete and recreate the namespace
+		// err = deleteNamespace(clientset, name)
+		// if err != nil {
+		// return err
+		// }
+	} else if !errors.IsNotFound(err) {
+		// Other error while getting namespace
+		return err
+	}
+
+	// Create the namespace if it doesn't exist
+	namespace := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
+	_, err = clientset.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+	return err
+}
+
+func deleteNamespace(clientset *kubernetes.Clientset, name string) error {
+	return clientset.CoreV1().Namespaces().Delete(context.Background(), name, metav1.DeleteOptions{})
+}
+
+func checkNameSpaceAnnotations(ns *v1.Namespace, expectedAnnotations []string) bool {
+	for _, annotation := range expectedAnnotations {
+		if ns.ObjectMeta.Annotations[annotation] != "true" {
+			return false
+		}
+	}
+	fmt.Println("Namespace annotations are correct!")
+	return true
+}
+func updateOperator(t *testing.T, clientSet *kubernetes.Clientset, deployment *appsV1.Deployment) bool {
+	opMutex.Lock() // Lock the mutex at the beginning
+	defer opMutex.Unlock()
+	var err error
+	args := deployment.Spec.Template.Spec.Containers[0].Args
+	// Attempt to get the deployment by name
+	deployment, err = clientSet.AppsV1().Deployments(amazonCloudwatchNamespace).Get(context.TODO(), amazonControllerManager, metav1.GetOptions{})
+	deployment.Spec.Template.Spec.Containers[0].Args = args
+	if err != nil {
+		t.Errorf("Failed to get deployment: %v\n", err)
+		return false
+	}
+
+	// Update the deployment
+	_, err = clientSet.AppsV1().Deployments(amazonCloudwatchNamespace).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+	if err != nil {
+		t.Errorf("Failed to update deployment: %v\n", err)
+		return false
+	}
+
+	fmt.Println("Deployment updated successfully!")
+	time.Sleep(70 * time.Second)
+	return true
+
+}
+func checkIfAnnotationExists(deploymentPods *v1.PodList, expectedAnnotations []string) bool {
+	for _, pod := range deploymentPods.Items {
+		for _, annotation := range expectedAnnotations {
+			if pod.ObjectMeta.Annotations[annotation] != "true" {
+				return false
+			}
+		}
+	}
+	fmt.Println("Annotations are correct!")
+	return true
+}
+func updateAnnotationConfig(deployment *appsV1.Deployment, jsonStr string) {
+	args := deployment.Spec.Template.Spec.Containers[0].Args
+	indexOfAutoAnnotationConfigString := findIndexOfPrefix("--auto-annotation-config=", args)
+	//if auto annotation not part of config, we will add it
+	if indexOfAutoAnnotationConfigString < 0 || indexOfAutoAnnotationConfigString >= len(deployment.Spec.Template.Spec.Containers[0].Args) {
```

Comment:
What is the point of checking `indexOfAutoAnnotationConfigString >= len(deployment.Spec.Template.Spec.Containers[0].Args` here? If  Doesn't `findIndexOfPrefix` function only return -1 when the index is not found?

Reply:
That's correct! I'll change that.